mozdownload/treeherder.py

```diff
@@ -21,48 +24,80 @@
     'win64': {'build_os': 'win', 'build_architecture': 'x86_64'},
 }
 
+TREEHERDER_HOST = 'treeherder.mozilla.org'
+
+
+class Treeherder(object):
+    """Wrapper class for TreeherderClient to ease the use of its API."""
+
+    def __init__(self, application, branch, platform,
+                 host=TREEHERDER_HOST, protocol=None):
+        """Create a new instance of the Treeherder class.
+
+        :param application: The name of the application to download.
+        :param branch: Name of the branch.
+        :param platform: Platform of the application.
+        :param host: The Treeherder host to make use of.
+        :param protocol: The protocol for the Treeherder host.
+        """
+        self.logger = logging.getLogger(__name__)
+
+        self.client = TreeherderClient(host=host, protocol=protocol)
+        self.application = application
+        self.branch = branch
+        self.platform = self.get_treeherder_platform(platform)
 
-def get_builds_from_revision(application, branch, revision, platform, job_type_name,
-                             debug_build=False):
-    """Retrieve builds from a given revision with the help of Treeherder."""
-    logger.info('Querying Treeherder to retrieve the list of builds for revision: %s' % revision)
+    def get_treeherder_platform(self, platform):
+        """Return the internal Treeherder platform identifier.
 
-    builds = set()
+        :param platform: Platform of the application.
+        """
+        return PLATFORM_MAP.get(platform, platform)
 
-    try:
-        client = TreeherderClient(protocol='https', host='treeherder.mozilla.org')
+    def query_builds_by_revision(self, revision, job_type_name='Build', debug_build=False):
+        """Retrieve build folders for a given revision with the help of Treeherder.
 
-        # Retrieve the option hash to filter for type of build (opt, and debug for now)
-        option_hash = None
-        for key, values in client.get_option_collection_hash().iteritems():
-            for value in values:
-                if value['name'] == ('debug' if debug_build else 'opt'):
-                    option_hash = key
+        :param revision: Revision of the build to download.
+        :param job_type_name: Name of the job to look for. Defaults to `Build`.
```

Comment:
I will add a list of possible options for now. Will be pushed with the next commit.